internal/http/check.go

```diff
@@ -0,0 +1,210 @@
+	if err != nil {
+		http.Error(w, "error reading request body", http.StatusBadRequest)
+		return
+	}
+	var job Job
+	if err = json.Unmarshal(body, &job); err != nil {
+		w.WriteHeader(500)
+		return
+	}
+
+	logger := c.Logger.WithFields(log.Fields{
+		"target":  job.Target,
+		"checkID": job.CheckID,
+	})
+	ctx := context.WithValue(r.Context(), "logger", logger)
+	checkState := &State{
+		state: agent.State{
+			Report: report.Report{
+				CheckData: report.CheckData{
+					CheckID:          job.CheckID,
+					StartTime:        job.StartTime, // TODO: Is this correct or should be time.Now()
+					ChecktypeName:    c.config.Check.CheckTypeName,
+					ChecktypeVersion: c.config.Check.CheckTypeVersion,
+					Options:          job.Options,
+					Target:           job.Target,
+				},
+				ResultData: report.ResultData{},
+			},
+		},
+	}
+
+	runtimeState := state.State{
+		ResultData:       &checkState.state.Report.ResultData,
+		ProgressReporter: state.ProgressReporterHandler(checkState.SetProgress),
+	}
+	logger.WithField("opts", job.Options).Info("Starting check")
+	err = c.checker.Run(ctx, job.Target, job.AssetType, job.Options, runtimeState)
+	c.checker.CleanUp(ctx, job.Target, job.AssetType, job.Options)
+	checkState.state.Report.CheckData.EndTime = time.Now()
+	elapsedTime := time.Since(startTime)
+	// If an error has been returned, we set the correct status.
+	if err != nil {
+		if errors.Is(err, context.Canceled) {
+			checkState.state.Status = agent.StatusAborted
+		} else if errors.Is(err, state.ErrAssetUnreachable) {
+			checkState.state.Status = agent.StatusInconclusive
+		} else if errors.Is(err, state.ErrNonPublicAsset) {
+			checkState.state.Status = agent.StatusInconclusive
+		} else {
+			logger.WithError(err).Error("Error running check")
+			checkState.state.Status = agent.StatusFailed
+			checkState.state.Report.Error = err.Error()
+		}
+	} else {
+		checkState.state.Status = agent.StatusFinished
+	}
+	checkState.state.Report.Status = checkState.state.Status
+
+	logger.WithFields(log.Fields{"seconds": elapsedTime.Round(time.Millisecond * 100).Seconds(), "state": checkState.state.Status}).Info("Check finished")
+
+	// Initialize sync point for the checker and the push state to be finished.
+	out, err := json.Marshal(checkState.state)
+	if err != nil {
+		logger.WithError(err).Error("error marshalling the check state")
+		http.Error(w, "error marshalling the check state", http.StatusInternalServerError)
+		return
+	}
+	w.Write(out)
+}
+
+// RunAndServe implements the behavior needed by the sdk for a check runner to
+// execute a check.
+func (c *Check) RunAndServe() {
+	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+		if c.shuttedDown == nil {
+			w.Header().Set("Content-Type", "application/json")
+			w.Write([]byte(`"OK"`))
+		} else {
+			// the server is shutting down
+			http.Error(w, "shuttingDown", http.StatusServiceUnavailable)
+		}
+	})
+	http.HandleFunc("/run", c.ServeHTTP)
```

Comment:
I'd create a new `http.ServeMux`, register the routes and assign it to the `http.Server.Handler` field.

Reply:
done 877c330718594bef1468b09846b8dbdfe4e3f388